src/widgets/canvas/circle.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{
+    style::Color,
+    widgets::canvas::{Painter, Shape},
+};
+
+/// Shape to draw a circle with a given center and radius and with the given color
+#[derive(Debug, Clone)]
+pub struct Circle {
+    pub x: f64,
+    pub y: f64,
+    pub radius: f64,
+    pub color: Color,
+}
+
+impl Shape for Circle {
+    fn draw(&self, painter: &mut Painter<'_, '_>) {
+        for angle in 0..360 {
+            let radians = f64::from(angle).to_radians();
+            let circle_x = self.radius.mul_add(radians.cos(), self.x);
+            let circle_y = self.radius.mul_add(radians.sin(), self.y);
+            if let Some((x, y)) = painter.get_point(circle_x, circle_y) {
+                painter.paint(x, y, self.color);
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::buffer::Buffer;
+    use crate::layout::Rect;
+    use crate::style::Color;
+    use crate::symbols::Marker;
+    use crate::widgets::canvas::{Canvas, Circle};
+    use crate::widgets::Widget;
+
+    #[test]
+    fn test_it_draws_a_circle() {
+        let mut buffer = Buffer::empty(Rect::new(0, 0, 10, 5));
+        let canvas = Canvas::default()
+            .paint(|ctx| {
+                ctx.draw(&Circle {
+                    x: 5.0,
+                    y: 2.0,
+                    radius: 5.0,
+                    color: Color::Reset,
+                });
+            })
+            .marker(Marker::Braille)
+            .x_bounds([-10.0, 10.0])
+            .y_bounds([-10.0, 10.0]);
+        canvas.render(buffer.area, &mut buffer);
+        let expected = Buffer::with_lines(vec![
+            "     ⢀⣠⢤⣀ ",
+            "    ⢰⠋  ⠈⣇",
+            "    ⠘⣆⡀ ⣠⠇",
+            "      ⠉⠉⠁ ",
+            "          ",
+        ]);
+        assert_eq!(buffer, expected);
```

Comment:
I've added an assert_buffer_eq in #161 that makes tests like this quite a bit easier to write (not merged yet, but it would let you just `let expected = Buffer::with_lines(vec!["          ";5]);` then copy paste the actual result. Saves having to deal with the buffer debug output craziness. It's not merged yet, take a look though and see if you think it would help unit tests generally.

Reply:
Yes that would be useful; I wrote a quick'n'dirty one-liner to collect the symbols to a string